imageio/plugins/pyav.py

```diff
@@ -0,0 +1,977 @@
+
+Backend Library: `PyAV <https://pyav.org/docs/stable/>`_
+
+This plugin that wraps pyAV, a set of pythonic bindings for the FFMPEG library.
+It is similar to our famous FFMPEG plugin, but offers a more performant and
+robut interface and aims to superseed the FFMPEG plugin in the future.
+
+
+Methods
+-------
+.. note::
+    Check the respective function for a list of supported kwargs and their
+    documentation.
+
+.. autosummary::
+    :toctree:
+
+    PyAVPlugin.read
+    PyAVPlugin.iter
+    PyAVPlugin.write
+    PyAVPlugin.properties
+    PyAVPlugin.metadata
+
+Pixel Formats (Colorspaces)
+---------------------------
+
+By default, this plugin converts the video into 8-bit RGB (called ``rgb24`` in
+ffmpeg). This is a useful behavior for many use-cases, but sometimes you may
+want to use the video's native colorspace or you may wish to convert the video
+into an entirely different colorspace. This is controlled using the ``format``
+kwarg. You can use ``format=None`` to leave the image in its native colorspace
+or specify any colorspace supported by FFMPEG as long as it is stridable, i.e.,
+as long as it can be represented by a single numpy array. Some useful choices
+include:
+
+- rgb24 (default; 8-bit RGB)
+- rgb48le (16-bit lower-endian RGB)
+- bgr24 (8-bit BGR; openCVs default colorspace)
+- gray (8-bit grayscale)
+- yuv444p (8-bit channel-first YUV)
+
+Further, FFMPEG maintains a list of available formats, albeit not as part of the
+narrative docs. It can be `found here
+<https://ffmpeg.org/doxygen/trunk/pixfmt_8h_source.html>`_ (warning: C source
+code).
+
+Filters
+-------
+
+On top of providing basic read/write functionality, this plugin allows you to
+use the full collection of `video filters available in FFMPEG
+<https://ffmpeg.org/ffmpeg-filters.html#Video-Filters>`_. This means that you
+can apply excessive preprocessing to your video before retrieving it as a numpy
+array or apply excessive post-processing before you encode your data.
+
+Filters come in two forms: sequences or graphs. Filter sequences are, as the
+name suggests, sequences of filters that are applied one after the other. They
+are specified using the ``filter_sequence`` kwarg. Filter graphs, on the other
+hand, come in the form of a directed graph and are specified using the
+``filter_graph`` kwarg.
+
+.. note::
+    All filters are either sequences or graphs. If all you want is to apply a
+    single filter, you can do this by specifying a filter sequence with a single
+    entry.
+
+A ``filter_sequence`` is a list of filters, each defined through a 2-element
+tuple of the form ``(filter_name, filter_parameters)``. The first element of the
+tuple is the name of the filter. The second element are the filter parameters,
+which can be given either as a string or a dict. The string matches the same
+format that you would use when specifying the filter using the ffmpeg
+command-line tool and the dict has entries of the form ``parameter:value``. For
+example::
```

Comment:
This is cool, but also leaning towards video processing, which is not our core business. We should be careful not to "pollute" the docs too much. I guess it's ok here though, since it's the last section in theses docs.

Reply:
Are there docs in pyav that we can delegate some of this to? Perhaps some of this would fit as an actual example, instead of a code sample in the docs.

Reply:
Unfortunately, no. Here are the docs on filters in pyav: https://pyav.org/docs/stable/api/filter.html . I worked out how to use them in pyav by digging through GH issues and source code. An item for the example page could, indeed, be nice, and I was already thinking of turning our GIF writing unit test into one. 

The downside of doing just an example is that the examples page is supposed to give various (general) usage examples of ImageIO, whereas the page here is supposed to discuss the features of the pyav plugin. The problem with that is that discussing/describing filter graphs is cumbersome, and a single example does more explaining than me writing a wall of text. That's essentially why I choose to put an example here. We could move them, but then we would need a decent (and clear) replacement for them here, because in the end we have to document available kwargs.

Reply:
Mmm, I guess we can then best leave it like this for now.